### ISSUE TITLE
ESLint_prettier_prettier rule inconsistencies

### ISSUE THREAD

**Issue by geoquant** [codacy/codacy-eslint#29]
The following issue was flagged in Codacy, but there are no warnings in the editor or code modifications when saving or running the cli. If the tools are equivalently configured, why are there different results?

<img width="669" alt="Screen Shot 2019-12-28 at 11 20 17 AM" src="https://user-images.githubusercontent.com/1237781/71547976-7b47e880-2964-11ea-8afe-9cb52df959d4.png">

**Comment by lolgab** [comment]
Hi @geoquant,
This happens because we updated production with the new version of our ESLint wrapper, which contains latest ESLint, but we didn't update CLI yet to the new version.
I'm currently working on it.
I'll let you know when the new CLI is out.

Thank you for pointing it out,
Lorenzo

**Comment by lolgab** [comment]
We updated our CLI now. The results should be consistent with the website.
CLI new version is 0.2.0.
I'm closing this, feel free to reopen it if you continue to have problems

**Comment by jnurkka** [comment]
@lolgab we still have issues with this. seems like codacy is not able to read the prettier config defined in our .eslintrc. Do you have any proposals how we can fix this?

**Comment by lolgab** [comment]
Hi @jnurkka,
Are you using the latest version of ESLint ?
Which [way are you using to pass options through `.eslintrc`](https://github.com/prettier/eslint-plugin-prettier#options)?
Because we already have a test case using `.prettierrc` config file and it works.
I'm trying to reproduce your problem.

**Comment by jnurkka** [comment]
`"eslint": { "version": "5.16.0",`
`"eslint-config-prettier": { "version": "3.6.0",`

In .eslintrc I use the following settings:
```
{
  "extends": [
    "airbnb",
    "prettier",
    "prettier/react"
  ],
  "plugins": [
    "import",
    "react",
    "prettier"
  ]
```

We only use the app.codacy.com web service, so haven't tried the CLI.

**Comment by lolgab** [comment]
You should update `eslint` to latest version to be [aligned with Codacy](https://github.com/codacy/codacy-eslint/blob/4013bbe9e7e9fa4c92ca4acf5157b9ed4bc4b211/package.json#L82).
We use `eslint-config-prettier` `6.10.0` and `eslint-plugin-prettier` `3.1.2`.
Probably with new versions even the way to configure prettier changed.. And our eslint is using the latest one.
Try to update these versions and let me know if something changes.

**Comment by lolgab** [comment]
Hey @jnurkka,
We did find a bug when relative paths are present inside the ESLint config files.
Do you have relative paths in it? In that case we are pushing a fix that let people use relative paths in config files (like `project: ["./tsconfig.json"]`)

**Comment by jnurkka** [comment]
Thanks for your response @lolgab! We dont have relative paths in our config files. However, our project has multiple applications inside it, each of which has their own .eslintrc config file. We tell vscode to read these with the following in .vscode/settings.json: "eslint.workingDirectories": ["./app1", "./app2", "./app3", etc..]

I noticed that in the web app you state that you will scan the root directory for a config file. This is probably the reason for our issues. Do you have a solution for us other than placing an eslint config file in the root of our project?

**Comment by lolgab** [comment]
We are pushing a change in the tool that will solve the problem with relative paths in config file https://github.com/codacy/codacy-eslint/pull/178 (by running eslint with the repository directory as $PWD). It will be online this week.
About the ESLint config file, I'm not completely sure, but it could work even if you don't have the `.eslintrc` file in the repository root.
If the website doesn't let you choose to use a custom config file because it can't find a config file in the root, you could try creating an empty `.eslintrc` file in the root only to fool the website to go in the custom config mode.
This because the underlying ESLint tool allows configuration files in the inner folders.

**Comment by jnurkka** [comment]
> If the website doesn't let you choose to use a custom config file because it can't find a config file in the root, you could try creating an empty `.eslintrc` file in the root only to fool the website to go in the custom config mode.
This because the underlying ESLint tool allows configuration files in the inner folders.

This didn't help. Still keep getting the same errors from prettier

**Comment by lolgab** [comment]
I'll let you know when we publish the new version with the fix I mentioned.
Sorry for the inconvenience.

**Comment by jnurkka** [comment]
> I'll let you know when we publish the new version with the fix I mentioned.
> Sorry for the inconvenience.

Alright, thanks

**Comment by lolgab** [comment]
The new version is published.
Could you try again to see if you still have problems? Did you update ESLint and prettier in your repository?

**Comment by jnurkka** [comment]
> The new version is published.
> Could you try again to see if you still have problems? Did you update ESLint and prettier in your repository?

The issue seems to indeed have been fixed, thanks a lot! :)

**Comment by lolgab** [comment]
Sorry for not answering, I just lost the notification.
I'm closing this then :)
Thank you very much for reporting.
Lorenzo

**Comment by jnurkka** [comment]
One more question @lolgab: codacy doesnt seem to find the correct eslint config file if it is not placed in the root folder of the repo. Also, adding an empty .eslintrc file into the root doesnt solve this issue. Do you have a solution for this?

**Comment by lolgab** [comment]
@jnurkka At the moment we support only config files in the root directory, sadly.

**Comment by lolgab** [comment]
I'm reopening this since other users are facing inconsistencies with prettier rules.

**Comment by nathan815** [comment]
I am also [facing this issue](https://app.codacy.com/gh/seeus-dev/seeus-app/pullRequest?prid=5222307). I get a lot of errors on Codacy.com from the ESLint Prettier rule but only 3 warnings when running `eslint` locally from my console. It doesn't seem to be adhering to my prettier or eslint config files.

**Comment by m0rph13** [comment]
We got this issue in multiple repositories, when will this irritating bug be fixed?

**Comment by lolgab** [comment]
> I am also facing this issue. I get a lot of errors on Codacy.com from the ESLint Prettier rule but only 3 warnings when running eslint locally from my console. It doesn't seem to be adhering to my prettier or eslint config files.

Tried to clone your repository and run:
```
$ npm install
$ npm run lint
```
And I get the same issues Codacy gives you.

**Comment by lolgab** [comment]
We have two prettier tests in [our integration tests](https://github.com/codacy/codacy-eslint/tree/master/docs/multiple-tests) and I added one right now. Still I can't reproduce the problem.

**Comment by nathan815** [comment]
Found that Codacy's prettier pattern is working, my config was just incorrect. Thank you @lolgab for the assistance 🙂

**Comment by lolgab** [comment]
I'm closing this issue since the prettier pattern seems to behave correctly.
If you are still experiencing it, feel free to comment.

**Comment by calmonr** [comment]
I'm currently having the same problem: https://github.com/calmonr/typegraphql-relay
![image](https://user-images.githubusercontent.com/49694957/103964419-2e335080-513a-11eb-9a2a-2d089f398edd.png)
![image](https://user-images.githubusercontent.com/49694957/103964438-3c816c80-513a-11eb-844b-0b4343433e69.png)

@lolgab

**Comment by merey90** [comment]
hey @lolgab , I am facing the same problem:
<img width="1643" alt="Screenshot 2021-05-20 at 10 06 57" src="https://user-images.githubusercontent.com/8048802/118942663-3bb3df00-b953-11eb-935c-bd5b810f0624.png">

eslintrc.
``` 
{
  "plugins": ["prettier", "cypress"],
  "extends": ["react-app", "prettier", "plugin:prettier/recommended"],
  "globals": {
    "cy": true
  },
  "rules": {
    "prettier/prettier": "error",
    "prefer-const": 1
  },
  "env": {
    "cypress/globals": true
  }
}
```
packages:
```
  "eslint": "^7.26.0",
  "eslint-config-prettier": "^8.3.0",
  "eslint-plugin-prettier": "^3.4.0",
  "prettier": "^2.3.0",
```

**Comment by merey90** [comment]
Downgrading `eslint` to version `7.25.0`(which is currently supported  by Codacy) didn't help =(

**Comment by merey90** [comment]
Downgrading the `prettier` to version `2.2.1` as it was before fixed the problem

**Comment by lolgab** [comment]
@calmonr I tried to rename the `.eslintrc` file to the canonical name `.eslintrc.json`, plus updating eslint-prettier-config, by following the [logs from Codacy](https://app.codacy.com/gh/lolgab/typegraphql-relay/commit?bid=23641113&cid=554992635) and now it seems all fine. Opened a PR to your repo here: https://github.com/calmonr/typegraphql-relay/pull/134

**Comment by merey90** [comment]
No, but here is the code:
```
import useSWR from 'swr';
import axios from 'axios';

const useRequest = (request, { initialData, ...config } = {}) => {
  const { data: response, error, isValidating, mutate } = useSWR(
    request && JSON.stringify(request),
    () => axios(request || {}),
    {
      revalidateOnFocus: false,
      ...config,
      initialData: initialData && {
        status: 200,
        statusText: 'InitialData',
        headers: {},
        data: initialData,
      },
    }
  );

  return {
    data: response?.data,
    response,
    error,
    isValidating,
    mutate,
  };
};

export default useRequest;
```
